replica/table.cc

```diff
@@ -1212,13 +1212,18 @@ void table::set_compaction_strategy(sstables::compaction_strategy_type strategy)
             cg.set_main_sstables(std::move(new_sstables));
         }
     };
-    compaction_group_sstable_set_updater cg_set_updater(*this, *_compaction_group, new_cs);
-
-    cg_set_updater.prepare(new_cs);
+    std::vector<compaction_group_sstable_set_updater> cg_sstable_set_updaters;
 
+    for (compaction_group_ptr& cg : compaction_groups()) {
+        compaction_group_sstable_set_updater updater(*this, *cg, new_cs);
+        updater.prepare(new_cs);
```

Comment:
Is it safe now to share a compaction strategy among multiple groups?

Reply:
Not yet.